src/Kernel32.Tests/storebanned/Kernel32Facts.cs

```diff
@@ -15,6 +15,7 @@
 
 public partial class Kernel32Facts
 {
+    private static unsafe Kernel32.THREAD_START_ROUTINE threadProc = new THREAD_START_ROUTINE(CreateThread_Test_ThreadMain);
```

Comment:
Does being static prevent it from being moved by the GC?

Could you solve the problem with `GC.KeepAlive` or pinning of the delegate rather than using a static?

Reply:
Pinning seems unnecessary here. It seems that the interop-marshaler doesn't pass on the address of either the delegate or the function itself; rather what gets passed on is the address of the thunk that gets generated on the fly, and this address is in fact non-relocatable. I couldn't find any documentation re: the last point (non-relocatability property of the thunk) , but a scan of SO and dotnet/runtime sources strongly hints that this is the case. 

From [callbackOnCollectedDelegate MDA](https://docs.microsoft.com/en-us/dotnet/framework/debug-trace-profile/callbackoncollecteddelegate-mda): 
>When delegates are marshaled as function pointers, the runtime allocates a thunk that does the transition from unmanaged to managed. This thunk is what the unmanaged code actually calls before the managed delegate is finally invoked

For prevention from being moved by the GC (hindsight being 20/20, I should have looked here for answers to the AV I was encountering earlier) - 

>The number one mistake made when using delegates as unmanaged function pointers is to allow them to be garbage collected before unmanaged code is finished using them. Premature collection of delegates can result in several different symptoms, depending on the implementation of the unmanaged code using the function pointers. For example, it could cause an access violation, or it could cause your callback methods to never be called while the rest of the program runs normally. An easy way to prevent collection is to assign the delegate reference to a static field,

> Nathan, Adam. .NET and COM: The Complete Interoperability Guide (Kindle Locations 14677-14681). Pearson Education (US). Kindle Edition. 

I figured that a `static` is simpler since the delegate was being reused across multiple tests. `GC.KeepAlive` or `GCHandle.Alloc` might also work, I think. 

With this context, would you rather we didn't clutter `Kernel32Facts` with more top-level statics and rather move it over to local instances in each test-case?

Reply:
What you have seems fine. I was mostly curious whether `GC.KeepAlive` would also work. My only concern for a test is that it can be used as a sample for how to accurately use an API, which yours does in one of a few likely workable fashions, so that's fine.